executor/adapter.go

```diff
@@ -100,3 +109,65 @@ func (a *statement) Exec(ctx context.Context) (ast.RecordSet, error) {
 		fields:   fs,
 	}, nil
 }
+
+// subqueryExecutor executes independent subquery.
+type subqueryExecutor struct {
+	ctx          context.Context
+	multipleRows bool
+	existRow     bool
+	err          error
+}
+
+func (e *subqueryExecutor) Enter(in ast.Node) (ast.Node, bool) {
+	switch in.(type) {
+	case *ast.ExistsSubqueryExpr:
+		e.existRow = true
+	case *ast.CompareSubqueryExpr:
+		e.multipleRows = true
+	case *ast.PatternInExpr:
+		e.multipleRows = true
+	}
+	return in, false
+}
+
+func (e *subqueryExecutor) Leave(in ast.Node) (ast.Node, bool) {
+	switch x := in.(type) {
+	case *ast.CompareSubqueryExpr:
+		e.multipleRows = false
+	case *ast.PatternInExpr:
+		e.multipleRows = false
+	case *ast.ExistsSubqueryExpr:
+		e.existRow = false
+	case *ast.SubqueryExpr:
+		if !x.UseOuterContext && x.SubqueryExec != nil {
+			rowCount := 2
```

Comment:
`Evaluator` can not be used directly.
I prefer  `copy on write`, so one doesn't affect the other.